src/krbd.cc

```diff
@@ -764,3 +764,58 @@ int krbd_showmapped(struct krbd_ctx *ctx, Formatter *f)
 {
   return dump_images(ctx, f);
 }
+
+int krbd_is_image_mapped(struct krbd_ctx *ctx, const char *poolname, 
+                       const char *imgname, const char *snapname,
+                       std::ostringstream &mapped_info) {
+  struct udev_enumerate *enm;
+  struct udev_list_entry *l;
+  struct udev *udev = ctx->udev; 
+  const char *mapped_id, *mapped_pool, *mapped_image, *mapped_snap;
+  int r = 1;
```

Comment:
I thought there is no need to use extra variable while only one variable can make it.
But you are right, the manner of image.is_snap_protected() make the code more readable.